iroha/src/lib.rs

```diff
@@ -296,6 +305,26 @@ impl Iroha {
         );
         Ok(())
     }
+
+    /// To wait for active peers(this is provisional code)
+    async fn wait_for_active_peers(&self, sorted_peers: &[PeerId]) -> Result<(), String> {
+        log::info!("Waiting for active peers");
+        for peer in sorted_peers.iter() {
+            let url = format!("http://{}{}", &peer.address, &uri::HEALTH_URI.to_string());
```

Comment:
I can't decide which is the best solution.
but the schema is required by `attohttpc::get`.
it returns this err `" Error: URL is missing a host, failed to send HTTP request to localhost:1339/health"`

Reply:
Would be great to use typed builder here to prevent future changes in `peer.address` to break code in production.

Reply:
Do you mean something like [url crate](https://docs.rs/url/2.2.0/url/) @humb1t ?

Reply:
Also I checked right now for me specifying the `localhost` work with attohttpc @MizukiSonoko

Reply:
Anyway try isahc first please @MizukiSonoko . Then let's see if it will have any errors.